src/OrleansCodeGenerator/SerializerGenerationManager.cs

```diff
@@ -121,6 +116,13 @@ internal bool RecordTypeToGenerate(Type t, Module module, Assembly targetAssembl
             return true;
         }
 
+        private static bool IsFieldInaccessibleForSerialization(Module module, Assembly targetAssembly, FieldInfo field)
+        {
+            return field.GetCustomAttribute<NonSerializedAttribute>() == null
+                   && !SerializationManager.HasSerializer(field.FieldType)
```

Comment:
do you think that order can affect us? If the serializer was manually crafted and registered in this assembly, would it already be in SerializationManager by the time codegen runs?
I assume it is, but just checking.